lib/decidim/direct_verifications/authorize_user.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Decidim
+  module DirectVerifications
+    class AuthorizeUser
+      def initialize(email, data, session, organization, instrumenter)
+        @email = email
+        @data = data
+        @session = session
+        @organization = organization
+        @instrumenter = instrumenter
+      end
+
+      def call
+        user = find_user
+
+        unless user
+          instrumenter.add_error :authorized, email
+          return
+        end
+
+        @authorization = find_or_create_authorization(user)
+        return unless valid_authorization?
+
+        Verification::ConfirmUserAuthorization.call(authorization, authorize_form(user), session) do
+          on(:ok) do
+            instrumenter.add_processed :authorized, email
+          end
+          on(:invalid) do
+            instrumenter.add_error :authorized, email
+          end
+        end
+      end
+
+      private
+
+      attr_reader :email, :data, :session, :organization, :instrumenter, :authorization
+
+      def valid_authorization?
+        !authorization.granted? || authorization.expired?
+      end
+
+      def find_user
+        User.find_by(email: email, decidim_organization_id: organization.id)
+      end
+
+      def find_or_create_authorization(user)
+        auth = Authorization.find_or_initialize_by(
+          user: user,
+          name: :direct_verifications
```

Comment:
I noticed the direct verifications controller allows passing in 

https://github.com/Platoniq/decidim-verifications-direct_verifications/blob/master/app/controllers/decidim/direct_verifications/verification/admin/direct_verifications_controller.rb#L23

and that can be specified in the request params:

https://github.com/Platoniq/decidim-verifications-direct_verifications/blob/master/app/controllers/decidim/direct_verifications/verification/admin/direct_verifications_controller.rb#L94-L96

but I found that `direct_verifications` is the only option

![Screenshot from 2021-04-01 19-58-32](https://user-images.githubusercontent.com/762088/113335157-d8c7a380-9324-11eb-9a41-b6bb5316f371.png)

Am I right in assuming this is just not needed and we can hardcode it?

Reply:
Nope, you can configure any other verification method in an initializer in order to check stats for that particular method.
What we could do is to hide this select if there's only one option.
https://github.com/Platoniq/decidim-verifications-direct_verifications#using-additional-verification-methods